benchmark/syclblas/main.cpp

```diff
@@ -8,40 +8,43 @@
 std::unique_ptr<cli_device_selector> cdsp;
 void free_device_selector() { cdsp.reset(); }
 
-// Declare the executorInstancePtr (it is already `extern` declared in utils.hpp
-// but it must be explicitly declared at least once.)
-ExecutorPtr Global::executorInstancePtr;
-
 int main(int argc, char** argv) {
+  // Read the command-line arguments
+  auto args = blas_benchmark::utils::parse_args(argc, argv);
+
+  // Initialize googlebench
+  benchmark::Initialize(&argc, argv);
+
   // Initialise the command line device selector in a unique pointer so that we
   // can register an `atexit` handler to delete the command line device
   // selector. If not, then if benchmark::Initialize calls exit() (for example,
   // if the flag `--help` is passed), the cds will not be freed before the sycl
   // runtime tries to exit, leading to an exception, as the cds will still hold
   // some sycl objects.
-  cdsp =
-      std::unique_ptr<cli_device_selector>(new cli_device_selector(argc, argv));
+  cdsp = std::unique_ptr<cli_device_selector>(new cli_device_selector(args));
   std::atexit(free_device_selector);
 
-  // Initialise googlebench
-  benchmark::Initialize(&argc, argv);
-
   // Create a queue from the device selector - do this after initialising
   // googlebench, as otherwise we may not be able to delete the queue before we
   // exit (if Initialise calls exit(0)), and dump some information about it
   cl::sycl::queue q = cl::sycl::queue(
       *cdsp.get(), {cl::sycl::property::queue::enable_profiling()});
-  benchmark::utils::print_queue_information(q);
+  blas_benchmark::utils::print_queue_information(q);
 
   // Create a sycl blas executor from the queue
-  Global::executorInstancePtr = std::unique_ptr<SyclExecutorType>(
+  ExecutorPtr executorInstancePtr = std::shared_ptr<SyclExecutorType>(
```

Comment:
This doesn't need to be a pointer any more - we can just create an executor, and then pass a pointer to it into blas_benchmark::create_benchmark, as the lifetime of the executor will be longer than the lifetime of the benchmarks.

Reply:
Done